File: extension/src/sidepanel/types/candidate.ts
export interface Education {
  school: string
  degree: string
  major: string
  graduation_year: number | null
}

export interface WorkExperience {
  company: string
  title: string
  duration_months: number
  description: string
  skills_used: string[]
}

export interface Candidate {
  name: string
  gender: string
  age: number
  years_of_experience: number
  current_title: string
  current_company: string
  education: Education[]
  work_experience: WorkExperience[]
  skills: string[]
  languages: string[]
  salary_expectation: string
  location: string
  summary: string
  source_raw_text: string
}

File: backend/app/models/candidate.py
from pydantic import BaseModel


class Education(BaseModel):
    school: str = ""
    degree: str = ""
    major: str = ""
    graduation_year: int | None = None


class WorkExperience(BaseModel):
    company: str = ""
    title: str = ""
    duration_months: int = 0
    description: str = ""
    skills_used: list[str] = []


class Candidate(BaseModel):
    name: str = ""
    gender: str = ""
    age: int = 0
    years_of_experience: int = 0
    current_title: str = ""
    current_company: str = ""
    education: list[Education] = []
    work_experience: list[WorkExperience] = []
    skills: list[str] = []
    languages: list[str] = []
    salary_expectation: str = ""
    location: str = ""
    summary: str = ""
    source_raw_text: str = ""
